### ISSUE TITLE
Option update in InsertCompletionSelect does not update the completion docstrings

### ISSUE THREAD

**Issue by danr** [mawww/kakoune#1859]
Source this beauty (beast?):

```kak
hook -group test global InsertChar .* %{
    try %{ decl completions test }
    set global test "%val{cursor_line}.%val{cursor_column}@%val{timestamp}:apa|apa docstring|apa:bepa||bepa"
    set global completers option=test
    echo no bepa: %opt{test}
}

hook -group test global InsertCompletionSelect .* %{
    set global test "%val{cursor_line}.%val{cursor_column}@%val{timestamp}:apa|apa docstring|apa:bepa|bepa docstring|bepa"
    echo yes bepa: %opt{test}
}
```

Enter insert mode and select something. 

Bepa now gets a docstring in the option, but it is not shown in the completion docstring area (which would be the expected).

**Comment by mawww** [comment]
Yeah, so this is happening because whenever a completion is selected, the option insert completer will not take updates from the option (as the user seems to be browsing in the existing list). I think this is a sane behaviour, but in the case you are in, the goal would be to update the docstring of the currently selected completion.

One way to do that would be to update the option and force reloading of the option menu and reselecting the same candidate.

**Comment by mawww** [comment]
Which I have been experimenting with, and seems tricky to get right, would be simpler to get the docstring for each completion candidate eagerly.

**Comment by danr** [comment]
I see, I hadn't thought about the complication that you would need to find the same candidate again. A complicated (but functional) way would be to diff the new option and if only docstrings have been updated it's immediate that the selected candidate is chosen.

I am leaning towards instead showing the dynamically obtained docstring in an `:info`, so I'm fine with closing this issue as a wontfix.
